src/software/gui/full_system/widgets/full_system_gui.cpp

```diff
@@ -11,23 +12,44 @@ FullSystemGUI::FullSystemGUI(
     std::shared_ptr<ThreadSafeBuffer<Rectangle>> view_area_buffer,
     std::shared_ptr<ThunderbotsConfig> config)
     : QMainWindow(),
-      main_widget(new MainWidget(config, this)),
+      main_widget(new Ui::AutogeneratedFullSystemMainWidget()),
       update_timer(new QTimer(this)),
       world_draw_functions_buffer(world_draw_functions_buffer),
       ai_draw_functions_buffer(ai_draw_functions_buffer),
       play_info_buffer(play_info_buffer),
       sensor_msg_buffer(sensor_msg_buffer),
       robot_status_buffer(robot_status_buffer),
       view_area_buffer(view_area_buffer),
-      update_timer_interval(Duration::fromSeconds(1.0 / 60.0))
+      most_recent_world_draw_function([](QGraphicsScene*) { return; }),
+      most_recent_ai_draw_function([](QGraphicsScene*) { return; })
 {
-    setCentralWidget(main_widget);
+    // Create a new widget that will contain all the autogenerated
+    // components defined in the .ui file. Note that because the
+    // setCentralWidget call will cause this class (the Qt QMainWindow)
+    // to take ownership of the widget and handle its deletion, we
+    // do not need to make this a class member and delete it ourself.
+    QWidget* central_widget = new QWidget(this);
+    main_widget->setupUi(central_widget);
+    setCentralWidget(central_widget);
 
-    connect(update_timer, &QTimer::timeout, this, &FullSystemGUI::updateGUI);
-    update_timer->start(static_cast<int>(update_timer_interval.getMilliseconds()));
+    // StrongFocus means that the MainWidget will more aggressively capture focus when
+    // clicked. Specifically, we do this so that when the user clicks outside of the
+    // QLineEdits used for Parameters, the QLineEdit will lose focus.
+    // https://www.qtcentre.org/threads/41128-Need-to-implement-in-place-line-edit-unable-to-get-lose-focus-of-QLineEdit
+    setFocusPolicy(Qt::StrongFocus);
+
+    setupRobotStatusTable(main_widget->robot_status_table_widget);
+    setupAIControls(main_widget, config);
+
+    // Update to make sure all layout changes apply nicely
+    update();
```

Comment:
If we didn't call this here, wouldn't it get called in the timer callback anyhow?

Reply:
yea turns out we don't really need it. Removed here and in other GUI classes.